### ISSUE TITLE
ERROR: not found, no name {test} in any of [], with proper name syntax

### ISSUE THREAD

**Issue by emptyopen** [taverntesting/tavern#366]
I copied the example test into a directory, and named it `test_basic.tavern.yaml`:

```
---
# Every test file has one or more tests...
test_name: Get some fake data from the JSON placeholder API

# ...and each test has one or more stages (e.g. an HTTP request)
stages:
  - name: Make sure we have the right ID

    # Define the request to be made...
    request:
      url: https://jsonplaceholder.typicode.com/posts/1
      method: GET

    # ...and the expected response code and body
    response:
      status_code: 200
      body:
        id: 1
``` 

Then on the commandline, running `tavern-ci test_basic.tavern.yaml` results in a successful test: 

![image](https://user-images.githubusercontent.com/18575100/58730019-c9314980-839f-11e9-8eaa-b66e828e0414.png)

But running any of `pytest test_basic.tavern.yaml`, `python3 -m pytest test_basic.tavern.yaml`, etc results in:

![image](https://user-images.githubusercontent.com/18575100/58730053-e2d29100-839f-11e9-8502-6ac972c9397f.png)

**Comment by michaelboulton** [comment]
Are you runnign this inside a virtualenv/pipenv? I've just tried locally and it works inside a virtualenv/pipenv, Tavern is also just a wrapper around Pytest https://github.com/taverntesting/tavern/blob/05c259cdf94786b568eaa027afbbf594c414cddd/tavern/core.py#L297 so hopefully there shouldn't be any difference in behaviour in test discovery

**Comment by emptyopen** [comment]
I tried running it inside of a virtualenv, same result. Should pytest be of some version? 
```
Administrators-MacBook-Pro-4:peatio-test-suite matt$ pytest --version
This is pytest version 4.5.0, imported from /usr/local/lib/python3.7/site-packages/pytest.py
```

**Comment by PaulCanada** [comment]
I also ran into this after upgrading tavern to 26.4, from 18.0.
This seems to be happening after version 20.0. As a result, I have to stick with version 20.0.

The format of my tests directory is:
`tests
----- Category
---------------TavernTesting
------------------------------'test_x.tavern.yaml'`

And I always get a ERROR: file not found: test_x.tavern.yaml

**Comment by michaelboulton** [comment]
I've tried with various combinations of versions of Python and inside and outside of a virtualenv end I can't reproduce this at all (also tried going before version 0.20 and after).

I did find this issue in another project https://github.com/pre-commit/pre-commit/issues/737 which leads me to believe it's something to do with the environment or the way paths are being handled/passed to pytest, but I don't have any suggestions on what the problem could be...

**Comment by borkfest** [comment]
Seeing the same issue with Ubuntu 18.04 LTS, Python 3.6.8, pytest-5.1.1, py-1.8.0, pluggy-0.12.0, tavern-0.28.0, outside and inside virtualenv

**Comment by borkfest** [comment]
Solved by downgrading pytest to 3.6.0 and naming all the test files to test_*.tavern.yaml

**Comment by michaelboulton** [comment]
I can't reproduce this and I think it's an environment-specific problem, so I'm closing this issue
